elements/rh-secondary-nav/test/rh-secondary-nav.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { RhSecondaryNavOverlay } from '../rh-secondary-nav-overlay';
+
+import { NAV } from './fixtures';
+
+function isDesktop() {
+  const matches = !!window.matchMedia('(min-width: 992px)').matches;
+  return matches;
+}
+
+let element: RhSecondaryNav;
+let mobileButton: HTMLButtonElement | null | undefined;
+let overlay: RhSecondaryNavOverlay | null | undefined;
+let dropdown: RhSecondaryNavDropdown | null;
+
+describe('<rh-secondary-nav>', async function() {
+  beforeEach(async function() {
+    element = await fixture<RhSecondaryNav>(NAV);
+    mobileButton = element.shadowRoot?.querySelector('button');
+    overlay = element.shadowRoot?.querySelector('rh-secondary-nav-overlay');
+    dropdown = element.querySelector('rh-secondary-nav-dropdown a');
+  });
+
+  it('should upgrade', async function() {
+    const klass = customElements.get('rh-secondary-nav');
+    expect(element)
+      .to.be.an.instanceOf(klass)
+      .and
+      .to.be.an.instanceOf(RhSecondaryNav);
+  });
+
+  it('should remove role="navigation" after upgrade', async function() {
+    expect(element.hasAttribute('role')).to.be.false;
+  });
+
+  it('should have an overlay set to hidden after upgrade', async () => {
+    const overlay: RhSecondaryNavOverlay | null | undefined = element.shadowRoot?.querySelector('rh-secondary-nav-overlay');
+    expect(overlay?.hasAttribute('hidden')).to.be.true;
+  });
+
+  it('lightdom passes the a11y audit', async () => {
+    await expect(element).to.be.accessible();
+  });
+
+  it('shadowdom passes the a11y audit', async function() {
+    await expect(element).shadowDom.to.be.accessible();
+  });
+
+  describe('dropdown is clicked', function() {
+    let event: SecondaryNavDropdownChangeEvent;
+    beforeEach(async function() {
+      setTimeout(() => dropdown?.click());
+      event = await oneEvent(element, 'change');
+    });
+
+    it('should fire a dropdown change event with expanded = true when a dropdown is clicked', async () => {
+      expect(event.expanded).to.be.true;
+    });
+
+    it('should fire a dropdown change event with expanded = false when a dropdown is clicked twice', async () => {
+      setTimeout(() => dropdown?.click());
+      event = await oneEvent(element, 'change');
+      expect(event.expanded).to.be.false;
+    });
+  });
+
+  describe('adjusting window size', function() {
+    describe('desktop screen', function() {
+      beforeEach(async function() {
+        await setViewport({ width: 992, height: 800 });
+        await element.updateComplete;
+        await aTimeout(100);
+        expect(isDesktop()).to.be.true;
+      });
+
+      it('should not have class is-mobile on shadow root nav', async () => {
+        await expect(element.shadowRoot?.querySelector('nav')?.classList.contains('is-mobile')).to.be.false;
+      });
+
+      it('lightdom passes the a11y audit', async function() {
+        await expect(element).to.be.accessible();
```

Comment:
i don't believe you need to run this on both the element and it's `shadowDom`. Just `expect(element).to.be.accessible()` is sufficient

Reply:
Fixed in bd4ee5b939bc72ff81264f416e1a76b5ac421e2d